pkg/server/api.go

```diff
@@ -283,8 +283,8 @@ func detectSpecVersionFromAcceptHeader(acceptHeader string) (*semver.Version, er
 
 	for _, header := range headers {
 		if header.MIMESubtype == "vnd.coreos.ignition+json" && header.SemVer != nil {
-			if !header.SemVer.LessThan(*v3_1) && header.SemVer.LessThan(*semver.New("4.0.0")) {
-				return v3_1, nil
+			if !header.SemVer.LessThan(*v3_2) && header.SemVer.LessThan(*semver.New("4.0.0")) {
```

Comment:
In both the API and the tests we need to also continue to handle 3.1 right?

Reply:
Ah yep, that was an oversight. I've got https://github.com/coreos/ign-converter/pull/19 up to add `v32tov31` down-translation. After that merges I'll bump the `ign-converter` vendor again & update to check for `v3_1` semver, down-translate in `server/api` & add tests for `3.1.0` in `server/api_test`.

Reply:
Ok, I think I've addressed this.

I've added back support for `v3_1` headers, added `3.2.0` ->` 3.1.0` translation, and re-added unit tests in `pkg/server/api_test`.